lib/components/viewers/stop-viewer.js

```diff
@@ -250,12 +250,21 @@ class StopViewer extends Component {
       stopData,
       stopViewerArriving,
       stopViewerConfig,
-      timeFormat
+      timeFormat,
+      transitOperators
     } = this.props
     const { scheduleView, spin } = this.state
     const hasStopTimesAndRoutes = !!(stopData && stopData.stopTimes && stopData.stopTimes.length > 0 && stopData.routes)
-    // construct a lookup table mapping pattern (e.g. 'ROUTE_ID-HEADSIGN') to an array of stoptimes
+    // construct a lookup table mapping pattern (e.g. 'ROUTE_ID-HEADSIGN') to
+    // an array of stoptimes
     const stopTimesByPattern = getStopTimesByPattern(stopData)
+    const routeComparator = coreUtils.route.makeRouteComparator(
+      transitOperators
+    )
+    const patternComparator = (patternA, patternB) => routeComparator(
+      patternA.route,
+      patternB.route
+    )
```

Comment:
This doesn't seem to have a fallback sort if the routes are the same. Should we have it fallback on pattern headsign or something along those lines?

Reply:
We can do that. I'd like to reuse a sorting helper that isn't exported from otp-ui. Can you please approve https://github.com/opentripplanner/otp-ui/pull/226 to make this possible?

Reply:
Approved.

Reply:
Fixed in d9410f90947481eaebe0a3c0d43c9ce0cff4af62 and e110f9aa9774b83ed8649bb7aeb76f533625a5ac